pytest_fixtures/authsource_fixtures.py

```diff
@@ -0,0 +1,207 @@
+"""Test class for Active Directory Feature
+
+:Requirement: Ldapauthsource
+
+:CaseAutomation: Automated
+
+:CaseLevel: Integration
+
+:CaseComponent: LDAP
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+import copy
+from airgun.session import Session
+
+from nailgun import entities
+from robottelo.config import settings
+from robottelo.constants import AUDIENCE_MAPPER
+from robottelo.constants import GROUP_MEMBERSHIP_MAPPER
+from robottelo.constants import LDAP_ATTR
+from robottelo.constants import LDAP_SERVER_TYPE
+from robottelo.datafactory import gen_string
+from robottelo.decorators import fixture
+from robottelo.rhsso_utils import create_mapper
+from robottelo.rhsso_utils import get_rhsso_client_id
+from robottelo.rhsso_utils import run_command
+
+
+@fixture(scope='module')
+def ldap_data():
+    return {
+        'ldap_user_name': settings.ldap.username,
+        'ldap_user_passwd': settings.ldap.password,
+        'base_dn': settings.ldap.basedn,
+        'group_base_dn': settings.ldap.grpbasedn,
+        'ldap_hostname': settings.ldap.hostname,
+    }
+
+
+@fixture(scope='module')
+def ipa_data():
+    return {
+        'ldap_ipa_user_name': settings.ipa.username_ipa,
+        'ipa_otp_username': settings.ipa.otp_user,
+        'ldap_ipa_user_passwd': settings.ipa.password_ipa,
+        'ipa_base_dn': settings.ipa.basedn_ipa,
+        'ipa_group_base_dn': settings.ipa.grpbasedn_ipa,
+        'ldap_ipa_hostname': settings.ipa.hostname_ipa,
+        'time_based_secret': settings.ipa.time_based_secret,
+    }
+
+
+@fixture(scope='function')
```

Comment:
Keep function scoped fixtures in test_module else make it module-level scope fixture. So that it can be created every time per module.

Applies for all the non-module/non-session/non-class scoped fixtures.

I can be wrong though, @mshriver what do you think on keeping function level scoped fixtures here ?

Reply:
I don't think there should be any restriction to the fixture scope used within these modules.

The fixture scope is relevant to its use within the framework and Satellite - there could be 10 test modules that contain test cases that need an authentication source configured on a function scoped level.